src/main/java/io/kestra/plugin/ai/memory/RedisMemory.java

```diff
@@ -0,0 +1,156 @@
+
+import java.io.IOException;
+import java.time.Duration;
+
+@Getter
+@SuperBuilder
+@NoArgsConstructor
+@JsonDeserialize
+@Schema(
+    title = "Chat Memory backed by Redis",
+    description = """
+        Persist chat memory in a Redis store using the memory ID as the key.
+        The Redis entry will expire after the provided TTL.
+        Ensure your Redis instance is reachable and configured via environment or plugin properties.
+        """
+)
+@Plugin(
+    examples = {
+        @Example(
+            title = "Use Redis-based chat memory for a conversation.",
+            code = """
+                id: chat-with-memory
+                namespace: company.team
+
+                inputs:
+                  - id: first
+                    type: STRING
+                    defaults: Hello, my name is John
+                  - id: second
+                    type: STRING
+                    defaults: What's my name?
+
+                tasks:
+                  - id: first
+                    type: io.kestra.plugin.ai.rag.ChatCompletion
+                    chatProvider:
+                      type: io.kestra.plugin.ai.provider.GoogleGemini
+                      modelName: gemini-2.5-flash
+                      apiKey: "{{ secret('GEMINI_API_KEY') }}"
+                    embeddingProvider:
+                      type: io.kestra.plugin.ai.provider.GoogleGemini
+                      modelName: gemini-embedding-exp-03-07
+                      apiKey: "{{ secret('GEMINI_API_KEY') }}"
+                    embeddings:
+                      type: io.kestra.plugin.ai.embeddings.KestraKVStore
+                    memory:
+                      type: io.kestra.plugin.ai.memory.RedisMemory
+                      host: localhost
+                      port: 6379
+                    systemMessage: You are an helpful assistant, answer concisely
+                    prompt: "{{inputs.first}}"
+
+                  - id: second
+                    type: io.kestra.plugin.ai.rag.ChatCompletion
+                    chatProvider:
+                      type: io.kestra.plugin.ai.provider.GoogleGemini
+                      modelName: gemini-2.5-flash
+                      apiKey: "{{ secret('GEMINI_API_KEY') }}"
+                    embeddingProvider:
+                      type: io.kestra.plugin.ai.provider.GoogleGemini
+                      modelName: gemini-embedding-exp-03-07
+                      apiKey: "{{ secret('GEMINI_API_KEY') }}"
+                    embeddings:
+                      type: io.kestra.plugin.ai.embeddings.KestraKVStore
+                    memory:
+                      type: io.kestra.plugin.ai.memory.RedisMemory
+                      host: localhost
+                      port: 6379
+                    systemMessage: You are an helpful assistant, answer concisely
+                    prompt: "{{inputs.second}}"
+                """
+        )
+    }
+)
+public class RedisMemory extends MemoryProvider {
```

Comment:
We decided against using suffix.
I'll rename the `KestraKvMemory` to `KestraKvStore` later (I would have some refacto anyway)

```suggestion
public class Redis extends MemoryProvider {
```